### PR TITLE
Feature/performance optimization

### PR DESCRIPTION
**Pull request by ice-hector**
Increase animated numbers performance twice. 

Example of total rendering duration:

old
13.082424879074097 
11.850545167922974 
12.436704397201538 

optimized
6.1792521476745605 
5.332279920578003 
6.095257043838501

### CODE REVIEW

File: src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import {IceLabel} from '@components/Labels/IceLabel';
 import {COLORS} from '@constants/colors';
-import {useAnimatedNumber} from '@hooks/useAnimatedNumber';
+import {AnimatedNumberText} from '@hooks/AnimatedNumber';
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx]: I think we need to put it to components, not hooks

File: src/screens/HomeFlow/Home/components/Pager/components/Engagement/index.tsx [hunk 2]
```diff
@@ -60,15 +74,15 @@ export const Engagement = memo(({darkMode}: Props) => {
...
+            <AnimatedMiningStreak />
           </Text>
         </View>
         <View style={[styles.titleValueContainer, styles.daysOff]}>
           <Text style={[styles.title, darkMode && commonStyles.darkText]}>
             {t('home.engagement.days_off')}
           </Text>
           <Text style={[styles.value, darkMode && commonStyles.darkText]}>
-            {animatedMiningSummaryRemainingFreeMiningSessions}
+            <AnimatedFreeMiningSessions />
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Pager/components/Engagement/index.tsx]: Why can we not use `AnimatedNumberText` directly here?

File: src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx (hunk 1 above)
Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx]: +1, and name the component and the file the same

File: src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx [hunk 3]
```diff
@@ -59,25 +56,41 @@ export const LevelCard = forwardRef(
       return <CardBaseSkeleton />;
     }
 
+    const header = ({style}: TextProps) => (
```
Comment (ice-orion) [inline src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx]: Let's name the component variables in CamelCase:
```
const Header = ...
```

File: src/screens/HomeFlow/Home/components/Overview/components/OnlineUsersHistory/index.tsx [hunk 4]
```diff
@@ -54,12 +49,16 @@ export const OnlineUsersHistory = () => {
     return <CardBaseSkeleton />;
   }
 
+  const header = ({style}: TextProps) => (
```
Comment (ice-orion) [inline src/screens/HomeFlow/Home/components/Overview/components/OnlineUsersHistory/index.tsx]: same about the naming

File: src/screens/HomeFlow/Home/components/Pager/components/Engagement/index.tsx (hunk 2 above)
Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Pager/components/Engagement/index.tsx]: +1, in this case wrapping components in `useCallback` doesn't make sense

File: src/screens/HomeFlow/Home/index.tsx [hunk 5]
```diff
@@ -40,6 +40,7 @@ export const Home = memo(() => {
           contentContainerStyle={tabBarOffset.current}
           showsVerticalScrollIndicator={false}>
           <Pager />
+
```
Comment (ice-orion) [inline src/screens/HomeFlow/Home/index.tsx]: We don't have indentations between components here, let's probably remove this one?

**Review by ice-atlas — APPROVED** [review]

File: src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx [hunk 6]
```diff
@@ -38,19 +44,23 @@ export const LevelCard = forwardRef(
       AchievementsSelectors.getLevelByUserId({userId}),
     );
 
-    const roleType = useSelector(
-      AchievementsSelectors.getRoleTypeByUserId({userId}),
+    const Header = useCallback(
```
Comment (ice-hades) [inline src/screens/HomeFlow/Home/components/Overview/components/LevelCard/index.tsx:47]: it is a renderHeader

File: src/screens/HomeFlow/Home/components/Overview/components/OnlineUsersHistory/index.tsx [hunk 7]
```diff
@@ -35,9 +35,11 @@ export const OnlineUsersHistory = () => {
   const isSplashHidden = useSelector(isSplashHiddenSelector);
   const totalActiveUsers = useSelector(totalActiveUsersSelector);
 
-  const animatedTotalActiveUsers = useAnimatedNumber(
-    totalActiveUsers,
-    formatNumber,
+  const Header = useCallback(
```
Comment (ice-hades) [inline src/screens/HomeFlow/Home/components/Overview/components/OnlineUsersHistory/index.tsx:38]: it is not a component, so a renderHeader

**Review by ice-hades — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**